help/help.c

```diff
@@ -779,7 +784,9 @@ static int help_mbox_open(struct Mailbox *m)
    * to avoid a useless re-caching of the same directory */
   if (access(C_HelpDocDir, F_OK) == 0)
   { /* ensure a proper path, especially without any trailing slashes */
-    mutt_str_replace(&C_HelpDocDir, NONULL(realpath(C_HelpDocDir, NULL)));
+    char *real = realpath(NONULL(C_HelpDocDir), NULL);
+    FREE(&C_HelpDocDir);
+    C_HelpDocDir = real;
```

Comment:
`realpath()` returns a copy and `mutt_str_replace()` **makes** a copy, so we don't need both.